bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -5277,7 +5277,6 @@ public void syncExec (Runnable runnable) {
  */
 public <T, E extends Exception> T syncCall(SwtCallable<T, E> callable) throws E {
 	Objects.nonNull(callable);
-	@SuppressWarnings("unchecked")
```

Comment:
Actually I'm not sure if this is really sane to be removed.

Reply:
I don't think so. When I place this kind of statement somewhere in SWT code, it (correctly) gives an "unchecked" warning. I don't know about any proper way to instantiate a generically typed array in a type-safe way in Java. Since this array is only used as a means to pass some value from within the lambda outside, what about replacing the array with something that is actually supposed to pass a reference, like an `AtomicReference`? It also provides `setPlain()` and `getPlain()` to avoid the synchronization overhead.

Reply:
I noticed that `unchecked` warnings were suppressed for the entire `Display` class and therefore this suppression was flagged as unnecessary. But I removed the class-level suppression so it came back. It was only suppressed at the class-level for mac-os.

But I also applied your suggestion and used AtomicReference using set/getPlain. Thanks for that.